lib/g_searcher_web/controllers/api/report_controller.ex

```diff
@@ -0,0 +1,30 @@
+defmodule GSearcherWeb.API.ReportController do
+  use GSearcherWeb, :controller
+
+  alias GSearcher.Reports
+  alias GSearcherWeb.ErrorHandler
+  alias GSearcherWeb.Validators.API.CreateReportParams
+  alias GSearcherWeb.Validators.ParamsValidator
+
+  def create(conn, params) do
+    %{id: user_id} = conn.assigns.user
+
+    with {:ok, %{csv: csv}} <-
+           ParamsValidator.validate(params, as: CreateReportParams),
+         {:ok, _report} <- Reports.create_report(user_id, csv.filename, csv.path) do
+      send_resp(conn, :no_content, "")
```

Comment:
small comment but instead of return `:no_content` status, we can return as `:created` status as we are creating a resource here